applications/StructuralMechanicsApplication/custom_processes/set_moving_load_process.cpp

```diff
@@ -0,0 +1,319 @@
+    {
+        int tmp = unsorted_conditions_v[i].Id();
+        auto& r_geom = unsorted_conditions_v[i].GetGeometry();
+        int tmp2 = r_geom.Points()[0].Id();
+        int tmp3 = r_geom.Points()[1].Id();
+        int tmp4 = 2;
+        double a = r_geom[0].X();
+        int b = 1 + 1;
+    }
+
+
+    std::vector<Condition> sorted_conditions;
+    std::vector<int> visited_indices;
+    GeometricalObject::GeometryType& r_geom_first = first_condition.GetGeometry();
+    std::set<unsigned long long> node_id_vector{ r_geom_first[0].Id(),r_geom_first[1].Id() };
+
+ 
+    while (visited_indices.size() != unsorted_conditions_v.size())
+    {
+        for (int i =0; i< unsorted_conditions_v.size(); i++)
+        {
+            Condition& r_cond = unsorted_conditions_v[i];
+            GeometricalObject::GeometryType& r_geom = r_cond.GetGeometry();
+
+
+            // check if current index is already added to sorted condition vector
+            if (!std::count(visited_indices.begin(), visited_indices.end(), i))
+            {
+                // check if geom has a shared node with previous geom
+                if (node_id_vector.find(r_geom.Points()[0].Id()) != node_id_vector.end() || node_id_vector.find(r_geom.Points()[1].Id()) != node_id_vector.end())
+                {
+                    if (sorted_conditions.size() == 0)
+                    {
+                        // check if both nodes of geom are equal to nodes in start element, only do this to add the first element in the sorted conditions vector
+                        if (node_id_vector.find(r_geom[0].Id()) != node_id_vector.end() && node_id_vector.find(r_geom.Points()[1].Id()) != node_id_vector.end())
+                        {
+                            node_id_vector = { r_geom[0].Id(),r_geom[1].Id() };
+                            sorted_conditions.push_back(r_cond);
+                            visited_indices.push_back(i);
+                        }
+                    }
+                    else
+                    {
+                        node_id_vector = { r_geom[0].Id(),r_geom[1].Id() };
+                        sorted_conditions.push_back(r_cond);
+                        visited_indices.push_back(i);
+                    }
+                }
+            }
+            
+        }
+    }
+
+    // swap first 2 conditions if nessesary
+    //r_geom = first_condition.GetGeometry();
+    int tmp = first_condition.Id();
+    for (int i = 0; i < sorted_conditions.size(); ++i)
+    {
+        int tmp2 = sorted_conditions[i].Id();
+        double tmp12 = sorted_conditions[i].GetGeometry()[0].X();
+        int tmp7 = 2;
+    }
+    
+
+    return sorted_conditions;
+}
+
+
+void SetMovingLoadProcess::ExecuteInitialize()
+{
+    KRATOS_TRY
+    mLoad = mParameters["load"].GetVector();
+    const double current_time = mrModelPart.GetProcessInfo().GetValue(TIME);
+
+    vector<int> direction = mParameters["direction"].GetVector();
```

Comment:
```suggestion
   const array_1d<double,3> direction = mParameters["direction"].GetVector();
```
These components are not necessarily unitary. Hence you should use a floating point array and then normalize it by taking its norm with `norm_2`.